flow/flow.go

```diff
@@ -894,6 +899,123 @@ func (f *Flow) newTransportLayer(packet *Packet, opts Opts) error {
 
 	return nil
 }
+func newDNSRecordFromLayersRecord(rr *layers.DNSResourceRecord) fl.DNSResourceRecord {
+	return fl.DNSResourceRecord{
+		Name:       string(rr.Name),
+		Type:       rr.Type.String(),
+		Class:      rr.Class.String(),
+		TTL:        rr.TTL,
+		DataLength: rr.DataLength,
+	}
+}
+func toFlowDNSResourceRecord(rr *layers.DNSResourceRecord) (fl.DNSResourceRecord, error) {
+	switch rr.Type {
+	case layers.DNSTypeA:
+	case layers.DNSTypeAAAA:
+		record := newDNSRecordFromLayersRecord(rr)
+		record.IP = rr.IP.String()
+		return record, nil
+	case layers.DNSTypeSOA:
+		record := newDNSRecordFromLayersRecord(rr)
+		record.SOA = &fl.DNSSOA{
+			MName:   string(rr.SOA.MName),
+			RName:   string(rr.SOA.RName),
+			Serial:  rr.SOA.Serial,
+			Refresh: rr.SOA.Refresh,
+			Retry:   rr.SOA.Retry,
+			Expire:  rr.SOA.Expire,
+			Minimum: rr.SOA.Minimum,
+		}
+		return record, nil
+	case layers.DNSTypeSRV:
+		record := newDNSRecordFromLayersRecord(rr)
+		record.SRV = &fl.DNSSRV{
+			Priority: rr.SRV.Priority,
+			Weight:   rr.SRV.Weight,
+			Port:     rr.SRV.Port,
+			Name:     string(rr.SRV.Name),
+		}
+		return record, nil
+	case layers.DNSTypeMX:
+		record := newDNSRecordFromLayersRecord(rr)
+		record.MX = &fl.DNSMX{
+			Preference: rr.MX.Preference,
+			Name:       string(rr.MX.Name),
+		}
+		return record, nil
+	case layers.DNSTypeTXT, layers.DNSTypeHINFO:
+		record := newDNSRecordFromLayersRecord(rr)
+		TXTs := make([]string, len(rr.TXTs))
+		for i, v := range rr.TXTs {
+			TXTs[i] = string(v)
+		}
+		record.TXTs = TXTs
+		return record, nil
+	case layers.DNSTypeNS:
+		record := newDNSRecordFromLayersRecord(rr)
+		record.NS = string(rr.NS)
+		return record, nil
+	case layers.DNSTypeCNAME:
+		record := newDNSRecordFromLayersRecord(rr)
+		record.CNAME = string(rr.CNAME)
+		return record, nil
+	case layers.DNSTypePTR:
+		record := newDNSRecordFromLayersRecord(rr)
+		record.PTR = string(rr.PTR)
+		return record, nil
+	}
+
+	return fl.DNSResourceRecord{}, errors.New("No Matching Type")
+}
+func dnsRecords(records []layers.DNSResourceRecord) []fl.DNSResourceRecord {
+	if len(records) > 0 {
+		holders := make([]fl.DNSResourceRecord, 0, len(records))
+		for _, v := range records {
+			if resource, err := toFlowDNSResourceRecord(&v); err == nil {
+				holders = append(holders, resource)
+			}
+		}
+		return holders
+	}
+	return nil
+}
+
+func (f *Flow) updateDNSLayer(packet *Packet) error {
+	layer := packet.Layer(layers.LayerTypeDNS)
+	d := layer.(*layers.DNS)
+	DNSToAppend := &fl.DNS{
```

Comment:
nit: please write it like this: `dnsToAppend`

Reply:
done